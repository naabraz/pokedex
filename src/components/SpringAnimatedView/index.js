// @flow

import React, { Component } from 'react'
import { Animated } from 'react-native'

type ISprinAnimated = {
  children: Object,
}

type State = {
  springValue: Object
}

class SpringAnimatedView extends Component<ISprinAnimated, State> {
  state = {
    springValue: new Animated.Value(0.3),
  }

  componentDidMount () {
    const { springValue } = this.state
    const springConfig = {
      toValue: 1,
      friction: 2,
    }

    springValue.setValue(0.3)
    Animated.spring(springValue, springConfig).start()
  }

  render () {
    const { springValue } = this.state
    const { children } = this.props

    return (
      <Animated.View style={{ transform: [{ scale: springValue }] }}>
        {children}
      </Animated.View>
    )
  }
}

export default SpringAnimatedView
