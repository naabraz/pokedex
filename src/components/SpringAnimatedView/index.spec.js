import React from 'react'
import renderer from 'react-test-renderer'

import SpringAnimatedView from '.'

describe('SpringAnimatedView component test', () => {
  it('Should render component correctly', () => {
    const animatedView = renderer.create(<SpringAnimatedView />).toJSON()
    expect(animatedView).toMatchSnapshot()
  })
})
