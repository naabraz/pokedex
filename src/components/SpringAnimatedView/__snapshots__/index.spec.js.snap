// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpringAnimatedView component test Should render component correctly 1`] = `
<View
  style={
    Object {
      "transform": Array [
        Object {
          "scale": 0.3,
        },
      ],
    }
  }
/>
`;
